trade-finance-manager-api/src/graphql/helpers/amendment.helpers.js

```diff
@@ -0,0 +1,189 @@
+const { DEALS } = require('../../constants');
+
+// returns the formatted amendment value and currency (without conversion)
+const amendmentChangeValueExportCurrency = (amendment) => {
+  const { currency, value: amendmentValue } = amendment;
+
+  if (currency && amendmentValue) {
+    return `${currency} ${formattedNumber(amendmentValue)}`;
+  }
+
+  return null;
+};
+
+// calculates new facility value in GBP
+const calculateNewFacilityValue = (exchangeRate, amendment) => {
+  const { currency, value: amendmentValue } = amendment;
+  let newValue;
+
+  if (currency && amendmentValue) {
+    // if already in GBP, just take the value
+    if (currency === CURRENCY.GBP) {
+      newValue = amendmentValue;
+    } else {
+      // if no exchange rate return null
+      if (!exchangeRate) {
+        return null;
+      }
+      const valueInGBP = amendmentValue * exchangeRate;
+      const totalDecimals = decimalsCount(valueInGBP);
+      // rounds value to 2dp
+      if (totalDecimals > 2) {
+        newValue = roundNumber(valueInGBP, 2);
+      } else {
+        newValue = valueInGBP;
+      }
+    }
+
+    return newValue;
+  }
+
+  return null;
+};
+
+// calculates new ukef exposure from amendment value
+const calculateUkefExposure = (facilityValueInGBP, coverPercentage) => {
+  let ukefExposure;
+
+  if (facilityValueInGBP && coverPercentage) {
+    // parse float as can have 2dp in facility value
+    const calculation = parseFloat(facilityValueInGBP, 10) * (coverPercentage / 100);
+    const totalDecimals = decimalsCount(calculation);
+
+    if (totalDecimals > 2) {
+      ukefExposure = roundNumber(calculation, 2);
+    } else {
+      ukefExposure = calculation;
+    }
+
+    return ukefExposure;
+  }
+
+  return null;
+};
+
+// checks for coverStartDate and returns based on deal-type as stored differently
+const dealTypeCoverStartDate = (facilitySnapshot) => {
+  const { coverStartDate } = facilitySnapshot;
+
+  // if exists - gef
+  if (coverStartDate) {
+    return coverStartDate;
+  }
+
+  let dateConstructed;
+
+  if (facilitySnapshot['requestedCoverStartDate-year'] && facilitySnapshot['requestedCoverStartDate-month'] && facilitySnapshot['requestedCoverStartDate-day']) {
+  // BSS stored as seperate year month day values
+    dateConstructed = new Date(
+      facilitySnapshot['requestedCoverStartDate-year'],
+      facilitySnapshot['requestedCoverStartDate-month'] - 1,
+      facilitySnapshot['requestedCoverStartDate-day'],
+    );
+  } else {
+    dateConstructed = new Date(parseInt(facilitySnapshot.requestedCoverStartDate, 10));
+  }
+
+  return dateConstructed;
+};
+
+// calculates new tenor from amendment coverEndDate and facility coverStartDate.  Requires external api call to calculate tenor
+const calculateAmendmentTenor = async (facilitySnapshot, amendment) => {
+  if (facilitySnapshot && facilitySnapshot?.ukefFacilityType && (facilitySnapshot?.coverStartDate || facilitySnapshot?.requestedCoverStartDate)
```

Comment:
1. can we put these conditions in a constant? there are a few conditions here. 
2. can we use `facilitySnapshot?.type`? we want to remove `ukefFacilityType`
3. you don't need `facilitySnapshot` because you already have `facilitySnapshot?.ukefFacilityType`
4. you don't need `amendment` because you already have `amendment?.coverEndDate`

Reply:
done